flink-runtime/src/main/java/org/apache/flink/streaming/api/operators/StreamOperatorStateContext.java

```diff
@@ -67,6 +67,12 @@ default boolean isRestored() {
      */
     InternalTimeServiceManager<?> internalTimerServiceManager();
 
+    /**
+     * Returns the internal timer service manager create by async state backend for the stream
+     * operator. This method returns null for non-keyed operators.
+     */
+    InternalTimeServiceManager<?> asyncInternalTimerServiceManager();
```

Comment:
Are there cases in which an operator needs to use async timer service and sync timer service simultaneously? I'm wondering if we could reduce `internalTimerServiceManager()` and `asyncInternalTimerServiceManager()` into one method, given that they have the same return type, and it might better help hiding implementation details from invokers.